src/Traits/HasArchives.php

```diff
@@ -81,6 +81,9 @@ public function getOriginalDirty(): array
      */
     public function saveArchive(): self
     {
+        //  Run the optional callback logic
+        $this->beforeArchiveCallback();
```

Comment:
this should be in `archiveWithData` or else it will be missed during the mass assignment archive step